### PR TITLE
Move notesubfoldertreewidget to its own class

### PR DESCRIPTION
**Pull request by Waqar144**

### CODE REVIEW

**Comment by pbek** [comment]
Seems to break some build processes...

**Comment by Waqar144** [comment]
Can you merge this please, if its acceptable? It blocks other things that I want to do.

File: src/mainwindow.ui [hunk 1]
```diff
@@ -770,7 +770,7 @@
              <rect>
               <x>0</x>
               <y>0</y>
-              <width>293</width>
+              <width>300</width>
```
Comment (pbek) [inline src/mainwindow.ui:773]: is this needed?

File: src/mainwindow.ui [hunk 2]
```diff
@@ -859,7 +859,7 @@ p, li { white-space: pre-wrap; }
      <x>0</x>
      <y>0</y>
      <width>1383</width>
-     <height>34</height>
+     <height>27</height>
```
Comment (pbek) [inline src/mainwindow.ui:862]: is this needed?

File: src/mainwindow.ui (hunk 1 above)
Comment (Waqar144) [reply src/mainwindow.ui:773]: ask qtcreator ¯\_(ツ)_/¯

File: src/mainwindow.ui (hunk 2 above)
Comment (Waqar144) [reply src/mainwindow.ui:862]: ¯\_(ツ)_/¯

File: src/mainwindow.ui (hunk 1 above)
Comment (pbek) [reply src/mainwindow.ui:773]: I'll like to keep such changes out... often enough they are messing with something on low resolutions...

**Comment by pbek** [comment]
Other than above... LGTM (without knowing what regressions it will cause 🙈 )

File: src/mainwindow.ui (hunk 1 above)
Comment (Waqar144) [reply src/mainwindow.ui:773]: They are not. And I doubt you even specified them by hand yourself.

Comment (pbek) [reply src/mainwindow.ui:773]: Every time I don't discard the changes QtCreator makes to the resolution of the settings dialog it is too big for small screens by default...

Comment (Waqar144) [reply src/mainwindow.ui]: done.

**Comment by Waqar144** [comment]
> Other than above... LGTM (without knowing what....

you don't have to merge, its just a proposal.

**Comment by pbek** [comment]
> its just a proposal.

is it? 🤣 

> Can you merge this please, if its acceptable? It blocks other things that I want to do.

😁 

Although it would be great if you would not force push to existing PRs... It's impossible to keep track of changes and review them and I don't even know that there was a change. 😅

**Comment by Waqar144** [comment]
The idea i am trying to implement is to separate the logic from the UI where possible. But it can't be done in one go, that'd be too error prone. So, I am trying to do something like this:

- separate out the individual components into their own classes
- make the components talk directly in cases where that's the origial intention instead of going via mainWindow
- Move out all the "globals/singletons" to a "kernel or engine" class

There is probably a lot more to this but often things become obvious once basics are done.

**Pull request merged**